### PR TITLE
Reset cursor of unacked packets in DataRequest

### PR DESCRIPTION
**Pull request by tekjar**

### CODE REVIEW

File: rumqttd/src/router/iobufs.rs [hunk 1]
```diff
@@ -56,7 +59,7 @@ pub struct Outgoing {
     /// Handle which is given to router to allow router to communicate with this connection
     pub(crate) handle: Sender<()>,
     /// The buffer to keep track of inflight packets.
-    inflight_buffer: VecDeque<(u16, FilterIdx, Cursor)>,
+    pub(crate) inflight_buffer: VecDeque<(u16, FilterIdx, Cursor)>,
```
Comment (tekjar) [inline rumqttd/src/router/iobufs.rs:62]: Remove `pub` if not necessary

File: rumqttd/src/router/iobufs.rs [hunk 2]
```diff
@@ -171,6 +174,17 @@ impl Outgoing {
 
         Some(())
     }
+
+    pub fn retrasmission_map(&self) -> HashMap<FilterIdx, Cursor> {
```
Comment (tekjar) [inline rumqttd/src/router/iobufs.rs:178]: Add unit tests

File: rumqttd/src/router/routing.rs [hunk 3]
```diff
@@ -367,6 +368,15 @@ impl Router {
                 .into_iter()
                 .for_each(|r| tracker.register_data_request(r));
 
+            debug!("{:?}, {:?} ", &tracker, retransmissions);
```
Comment (tekjar) [inline rumqttd/src/router/routing.rs:371]: Remove this

File: rumqttd/src/router/routing.rs [hunk 4]
```diff
@@ -367,6 +368,15 @@ impl Router {
...
+
+            for request in tracker.data_requests.iter_mut() {
+                if let Some(cursor) = retransmissions.get(&request.filter_idx) {
+                    request.cursor = *cursor;
+                }
+            }
+
+            debug!("{:?}, {:?} ", &tracker, retransmissions);
```
Comment (tekjar) [inline rumqttd/src/router/routing.rs:379]: Remove this

File: rumqttd/src/segments/segment.rs [hunk 5]
```diff
@@ -135,11 +142,14 @@ mod tests {
         segment.push(Bytes::from_static(b"test9"));
         assert_eq!(segment.len(), 9);
 
-        let mut out: Vec<Bytes> = Vec::new();
-        let _ = segment.readv(0, 2, &mut out).unwrap();
```
Comment (tekjar) [inline rumqttd/src/segments/segment.rs:139]: See if we can add any more unit tests

Comment (henil) [reply rumqttd/src/segments/segment.rs:139]: i don't think we can add any more unit test here, `Segments` is just a wrapper around vec with some more metadata. `absolute_offset` stays constant for a particular segment and `size` is tested for by unit tests in `segments/mod.rs`

**Pull request merged**